pkg/providers/amifamily/ami.go

```diff
@@ -160,9 +161,14 @@ func (p *DefaultProvider) amis(ctx context.Context, queries []DescribeImageQuery
 	}
 	images := map[uint64]AMI{}
 	for _, query := range queries {
-		if err = p.ec2api.DescribeImagesPagesWithContext(ctx, query.DescribeImagesInput(), func(page *ec2.DescribeImagesOutput, _ bool) bool {
+		paginator := ec2.NewDescribeImagesPaginator(p.ec2api, query.DescribeImagesInput())
+		for paginator.HasMorePages() {
+			page, err := paginator.NextPage(ctx)
+			if err != nil {
+				return nil, fmt.Errorf("describing images, %w", err)
+			}
 			for _, image := range page.Images {
-				arch, ok := v1.AWSToKubeArchitectures[lo.FromPtr(image.Architecture)]
+				arch, ok := v1.AWSToKubeArchitectures[string(image.Architecture)]
```

Comment:
I do not think I can change the type because the 2 times its used is 2 different types [ec2types.ArchitectureType](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/ec2@v1.181.0/types#ArchitectureType:~:text=stable%20across%20updates.-,type%20ArchitectureType,-%C2%B6) and [ec2types.ArchitectureValues](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/ec2@v1.181.0/types#ArchitectureType). But I am actually not sure why they are separate types in v2 because they look to be the exact same.